includes/class-wcs-cart-renewal.php

```diff
@@ -494,33 +494,49 @@ public function get_cart_item_from_session( $cart_item_session_data, $cart_item,
 	 * Returns address details from the renewal order if the checkout is for a renewal.
 	 *
 	 * @param string $value Default checkout field value.
-	 * @param string $key The checkout form field name/key
+	 * @param string $key   The checkout form field name/key.
+	 *
 	 * @return string $value Checkout field value.
 	 */
 	public function checkout_get_value( $value, $key ) {
 
-		// Only hook in after WC()->checkout() has been initialised
-		if ( $this->cart_contains() && did_action( 'woocommerce_checkout_init' ) > 0 ) {
+		// Only hook in after WC()->checkout() has been initialised.
+		if ( ! $this->cart_contains() || did_action( 'woocommerce_checkout_init' ) <= 0 ) {
+			return $value;
+		}
 
-			// Guard against the fake WC_Checkout singleton, see https://github.com/woocommerce/woocommerce-subscriptions/issues/427#issuecomment-260763250
-			remove_filter( 'woocommerce_checkout_get_value', array( &$this, 'checkout_get_value' ), 10 );
+		// Get the most specific order object, which will be the renewal order for renewals, initial order for initial payments, or a subscription for switches/resubscribes.
+		$order = $this->get_order();
 
-			if ( is_callable( array( WC()->checkout(), 'get_checkout_fields' ) ) ) { // WC 3.0+
-				$address_fields = array_merge( WC()->checkout()->get_checkout_fields( 'billing' ), WC()->checkout()->get_checkout_fields( 'shipping' ) );
-			} else {
-				$address_fields = array_merge( WC()->checkout()->checkout_fields['billing'], WC()->checkout()->checkout_fields['shipping'] );
-			}
+		if ( ! $order ) {
+			return $value;
+		}
 
-			add_filter( 'woocommerce_checkout_get_value', array( &$this, 'checkout_get_value' ), 10, 2 );
+		// Guard against the fake WC_Checkout singleton, see https://github.com/woocommerce/woocommerce-subscriptions/issues/427#issuecomment-260763250
+		remove_filter( 'woocommerce_checkout_get_value', array( &$this, 'checkout_get_value' ), 10 );
```

Comment:
Because the code has changed to use `WC()->countries->get_address_fields()` do we still need these remove/add filters? Is it more of a precaution?

Reply:
yeah I think I can remove those now. I've removed that in https://github.com/Automattic/woocommerce-subscriptions-core/pull/596/commits/2b864be82ba25bb03f978a5ecb4737c57a84e904 and confirmed there's no infinite loop.